### PR TITLE
fix: backend error messages and dbt cloud adapterType

### PR DESCRIPTION
**Pull request by AdiGajbhiye**
## Overview

- handled backend api error messages
- fix dbt cloud adapterType

### Problem

Describe the problem you are solving. Mention the ticket/issue if applicable.

### Solution

Describe the implemented solution. Add external references if needed.

### Screenshot/Demo

A picture is worth a thousand words. Please highlight the changes if applicable.

### How to test

- Steps to be followed to verify the solution or code changes
- Mention if there is any settings configuration added/changed/deleted

## Checklist

- [ ] I have run this code and it appears to resolve the stated issue
- [ ] `README.md` updated and added information about my change

----

> [!IMPORTANT]
> Improves error handling in `AltimateRequest` and fixes adapter type initialization in `DBTCloudProjectIntegration`.
> 
>   - **Error Handling**:
>     - In `AltimateRequest` class in `altimate.ts`, `fetch` method now attempts to parse `textResponse` as JSON to extract `detail` field for `APIError` message.
>     - If JSON parsing fails, falls back to using `textResponse` directly in `APIError` message.
>   - **DBT Cloud Integration**:
>     - In `DBTCloudProjectIntegration` class in `dbtCloudIntegration.ts`, changed adapter type check from `!this.adapterType` to `this.adapterType === "unknown"` to ensure correct initialization of adapter type.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=AltimateAI%2Fvscode-dbt-power-user&utm_source=github&utm_medium=referral)<sup> for 7e4e4b577b34690aad52f3d375594a3a6a68f02d. It will automatically update as commits are pushed.</sup>

----

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved error messages for API response failures to provide clearer context.
	- Enhanced error handling for specific HTTP status codes (401, 403, 404, 402, 429) to improve user feedback.
	- Updated error handling in the DBT Cloud integration to provide clearer messages during initialization failures.
	- Refined logic for checking adapter type in the DBT Cloud integration for better accuracy.

### CODE REVIEW

File: src/dbt_client/dbtCloudIntegration.ts [hunk 1]
```diff
@@ -176,7 +176,7 @@ export class DBTCloudProjectIntegration
   private targetPath?: string;
   private version: number[] | undefined;
   private projectName: string = "unknown_" + crypto.randomUUID();
-  private adapterType: string = "unknown";
```
Comment (mdesmet) [inline src/dbt_client/dbtCloudIntegration.ts:179]: Is this required, this will change azure logs?

File: src/altimate.ts [hunk 2]
```diff
@@ -647,8 +647,12 @@ export class AltimateRequest {
         status: response.status,
         textResponse,
       });
+      let jsonResponse: any;
+      try {
+        jsonResponse = JSON.parse(textResponse);
+      } catch {}
       throw new APIError(
-        `Could not process request, server responded with ${response.status}: ${textResponse}`,
+        `Could not process request, server responded with ${response.status}: ${jsonResponse?.detail || textResponse}`,
```
Comment (mdesmet) [inline src/altimate.ts:655]: What is this solving or providing opposed to the earlier code?

Comment (AdiGajbhiye) [reply src/altimate.ts:655]: In some cases, the message shown in error notification contains json. Now I am just passing relevant details from backend error message.

image

Comment (mdesmet) [reply src/altimate.ts:655]: What if detail doesn't exist but the repsonse is json?

Comment (AdiGajbhiye) [reply src/altimate.ts:655]: this shouldn't happen as fastapi gives error response in this format only.
@suryaiyer95 correct me if i am wrong.

**Review by mdesmet — APPROVED** [review]

**Pull request merged**